signwriting_animation/diffusion/tests/test_length_prediction.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+import torch
+from torch.utils.data import DataLoader
+from signwriting_animation.data.data_loader import DynamicPosePredictionDataset, get_num_workers
+from signwriting_animation.diffusion.core.models import SignWritingToPoseDiffusion
+from pose_format.torch.masked.collator import zero_pad_collator
+import matplotlib.pyplot as plt
+import os
+
+@pytest.mark.parametrize("batch_size", [4])
+def test_length_prediction_on_real_data(batch_size):
+    data_dir = "/scratch/yayun/pose_data"
+    csv_path = "/data/yayun/signwriting-animation/data.csv"
+
+    dataset = DynamicPosePredictionDataset(
```

Comment:
you could create a "mock" dataset in order to test just this rather than the entire process